runtime/test/particle-api-test.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+const Manifest = require('../manifest.js');
+const assert = require('chai').assert;
+const util = require('./test-util.js');
+const viewlet = require('../viewlet.js');
+const Arc = require("../arc.js");
+const MessageChannel = require("../message-channel.js");
+const InnerPec = require("../inner-PEC.js");
+const Loader = require("../loader.js");
+
+describe('particle-api', function() {
+  it('contains a constructInnerArc call', async () => {
+    let registry = {};
+    let loader = new class extends Loader {
+      loadResource(path) {
+        return {
+          manifest: `
+            schema Result
+              normative
+                Text value
+
+            particle P in 'a.js'
+              P(out Result result)
+
+            recipe
+              use 'test:1' as view0
+              P
+                result -> view0
+
+          `,
+          'a.js': `
+            "use strict";
+
+            defineParticle(({Particle}) => {
+              return class P extends Particle {
+                async setViews(views) {
+                  let result = await this.constructInnerArc();
+                  var view = views.get('result');
+                  result = (result == undefined ? 'fail' : 'success');
```

Comment:
nit - if result is null, it will also return 'success', is that intended?

Reply:
Nothing will return null just yet. I'll keep adjusting these tests with further code changes though.